src/components/text-editor/text-editor.tsx

```diff
@@ -41,12 +100,43 @@ export class TextEditor {
      * Dispatched when a change is made to the editor
      */
     @Event()
-    private change: EventEmitter<{ html: string }>;
+    public change: EventEmitter<{ html: string }>;
+
+    private disabledKey: PluginKey;
+    private placeholderKey: PluginKey;
+    private readonlyKey: PluginKey;
+
+    @Watch('disabled')
+    public watchDisabled() {
+        this.view.dispatch(
+            this.view.state.tr.setMeta(this.disabledKey, this.disabled),
+        );
+    }
+
+    @Watch('placeholder')
+    public watchPlaceholder() {
+        this.view.dispatch(
+            this.view.state.tr.setMeta(this.placeholderKey, this.placeholder),
+        );
+    }
+
+    @Watch('readonly')
+    public watchReadonly() {
+        this.view.dispatch(
+            this.view.state.tr.setMeta(this.readonlyKey, this.readonly),
+        );
+    }
```

Comment:
Not sure why you are doing this. But I think it's better to just return `limel-markdown` when `readonly="true"`

Reply:
I'm not sure, but I worry that we then need complexity in some other place to hide and unhide the editor. I like this 
separation of concerns, as here we encapsulate editor-specific behaviour as an editor plugin, instead of doing it from within the consumer which I think should mostly just mediate between editor state changes and render the editor. Any thoughts @adrianschmidt ?